dashboard/config/libraries/NativeSpriteLab.interpreted.js

```diff
@@ -167,36 +170,106 @@ function setupSim(
     };
   }
 
+  // We don't allow certain list functions in the interpreter apparently, so writing my own.
+  function removeFromList(list, item) {
+    var newList = [];
+    for (var removeCounter = 0; removeCounter < list.length; removeCounter++) {
+      if (list[removeCounter] != item) {
+        newList.push(list[removeCounter]);
+      }
+    }
+    return newList;
+  }
+  function listIncludes(list, item) {
+    for (var includesCounter = 0; includesCounter < list.length; includesCounter++) {
+      if (list[includesCounter] == item) {
+        return true;
+      }
+    }
+    return false;
+  }
+
+  // Initialize sprites
   var counter_i = 0;
-  for (counter_i = 0; counter_i < s3number; counter_i++) {
-    makeNewSpriteAnon(s3costume, randomLocation());
+  for (counter_i = 0; counter_i < s1number; counter_i++) {
+    makeNewSpriteAnon(s1costume, randomLocation());
   }
-  setProp({costume: s3costume}, 'scale', 50);
+  addBehaviorSimple({costume: s1costume}, new Behavior(movementBehavior(s1speed)));
 
   for (counter_i = 0; counter_i < s2number; counter_i++) {
     makeNewSpriteAnon(s2costume, randomLocation());
   }
   addBehaviorSimple({costume: s2costume}, new Behavior(movementBehavior(s2speed)));
 
-  for (counter_i = 0; counter_i < s1number; counter_i++) {
-    makeNewSpriteAnon(s1costume, randomLocation());
+  for (counter_i = 0; counter_i < s3number; counter_i++) {
+    makeNewSpriteAnon(s3costume, randomLocation());
   }
-  addBehaviorSimple({costume: s1costume}, new Behavior(movementBehavior(s1speed)));
+  setProp({costume: s3costume}, 'scale', 50);
+
+
+  // We want to be able to randomize which sprite gets the point in the case of a tie. So the approach here is
+  // to use checkTouching() to detect all the collisions, but delay until the next frame to actually give a point
+  // and remove the s3 sprite. I'm using World.s1ToCollect and World.s2ToCollect to keep track of which sprite ids
+  // will be removed the following frame. Then if there's a tie (ie. s2 wants to remove a sprite id that's already
+  // in World.s1ToCollect or vice versa), use randomNumber(0,1) to give the sprite a 50-50 chance of "stealing" the
+  // s3 sprite.
+  // Then to actually remove the s3 sprites and keep track of the score, I'm using a behavior. I have the behavior
+  // attached to sprite id 0, but it doesn't really matter which sprite (as long as it's not an s3 sprite that
+  // might get deleted). A behavior is just the simplest way to add a snippet of code that will get executed each frame.
+  // The behavior basically just tallies the score and deletes the sprites from the previous frame, then checks
+  // whether the simulation should end. The one-frame delay isn't really noticeable since frames are so fast.
```

Comment:
Nit - can this be a multi-line comment? I find those easier to read for longer chunks of text.